base-api-client/src/main/java/com/cloudesire/platform/apiclient/baseclient/AbstractCloudesireClient.java

```diff
@@ -23,13 +23,26 @@
     protected ObjectMapper mapper;
 
     protected AbstractCloudesireClient( ObjectMapper mapper, String baseUrl, String userAgent )
+    {
+        this(
+                mapper,
+                baseUrl,
+                userAgent,
+                new TimeoutConfig.Builder()
+                        .setConnectTimeout( 30, TimeUnit.SECONDS )
+                        .setReadTimeout( 60, TimeUnit.SECONDS )
+                        .build()
+        );
+    }
+
+    protected AbstractCloudesireClient( ObjectMapper mapper, String baseUrl, String userAgent, TimeoutConfig timeoutConfig )
     {
         this.baseUrl = baseUrl;
         this.mapper = mapper;
 
         okhttpClientBuilder = new OkHttpClient.Builder()
-                .connectTimeout( 30, TimeUnit.SECONDS )     // connect timeout
-                .readTimeout( 60, TimeUnit.SECONDS );       // socket timeout
+                .connectTimeout( timeoutConfig.getConnectTimeout(), TimeUnit.MILLISECONDS )     // connect timeout
+                .readTimeout( timeoutConfig.getReadTimeout(), TimeUnit.MILLISECONDS );          // socket timeout
```

Comment:
non mettere due cose strettamente legate fra loro in due posti diversi, che se poi rifattorizzi non ti ricordi che qui avevi hardcodato millisecondi e che rompi tutti se cambi la classe
```
                .readTimeout( timeoutConfig.getReadTimeout(), timeoutConfig.getDefaultTimeUnit);          // socket timeout
```